libfreerdp/core/transport.c

```diff
@@ -371,6 +371,35 @@ BOOL transport_accept_nla(rdpTransport* transport)
 	return TRUE;
 }
 
+void transport_bio_error_log(char *biofunc, rdpTransport* transport)
+{
+	unsigned long sslerr;
+	char *buf;
+
+	if (!transport->frontBio)
+	{
+		WLog_ERR(TAG, "%s called with transport->frontBio null", biofunc);
+		return;
+	}
+
+	if (ERR_peek_error() == 0)
+	{
+		WLog_ERR(TAG, "%s returned system error %d: %s", biofunc, errno, strerror(errno));
+		return;
+	}
+
+	buf = malloc(120);
+	if (buf)
+	{
+		while((sslerr = ERR_get_error()))
+		{
+			ERR_error_string(sslerr, buf);
```

Comment:
@akallabeth I have watched the doc and it says that a buffer of at least 120 bytes is enough for ERR_error_string.